apollo-router-core/src/plugin.rs

```diff
@@ -221,3 +234,53 @@ macro_rules! register_plugin {
         }
     };
 }
+
+#[derive(Clone)]
+pub struct Handler {
```

Comment:
I am wondering if the handler type should exist. It looks like it just buffers a box service of the correct type.

Reply:
After investigations and different implementations I think it's cleaner to keep the `Handler` type. If you look into the `Service` implementation for `Handler` it specifies `type Future = ResponseFuture<BoxFuture<'static, Result<Self::Response, Self::Error>>>;`  With `warp` we need this `'static` lifetime. In `BoxService` it doesn't enforce the `'static` lifetime then it means I would have to not use `BoxService` and instead use `Box<dyn Service....>>`. After reading the code of tower it's not so uncommon to have this kind of pattern (cf `ServiceFn`)

Reply:
OK, let's merge as is. I may also have a look later.